### PR TITLE
Week 4 - Authorization

### PR DESCRIPTION
**Pull request by malcolmrjones**
Implemented authorization for the comments feature. Uses AppEngine's User API. Also update the styling of the comment section

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  private final String content;
+
+  public Comment(long id, long timestamp, String authorName, String authorEmail, String content) {
+    this.id = id;
+    this.timestamp = timestamp;
+    this.authorName = authorName;
+    this.authorEmail = authorEmail;
+    this.content = content;
+  }
+}
```
Comment (brucezheng) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: Nit: Add newline

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -30,20 +49,64 @@ function fetchComments() {
...
   //Remove all children of comment list to prevent dusplicate display
   while (commentList.firstChild) {
     commentList.removeChild(commentList.firstChild);
   }
 
   for (comment of comments) {
-    const commentListItem = document.createElement('li');
-    commentListItem.innerText = comment;
+    let displayName = comment["authorName"];
+    if (displayName === "") displayName =  comment["authorEmail"];
```
Comment (brucezheng) [inline portfolio/src/main/webapp/script.js]: Nit: Extra space between ```=``` and ```comment```

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -30,20 +49,64 @@ function fetchComments() {
...
+      Number(comment["timestamp"]), comment["content"]);
     commentList.appendChild(commentListItem);
   }
 }
 
+/**
+ * Creates the DOM element for a comment
+ * @param {!string} author 
+ * @param {!number} timestamp
+ * @param {!string} content
```
Comment (brucezheng) [inline portfolio/src/main/webapp/script.js]: For ```string``` and ```number```, you do not need to use ```!``` to indicate non-nullity.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -30,20 +49,64 @@ function fetchComments() {
...
+
+  const commentDate = new Date(timestamp);
+
+  commentItemAuthor.innerHTML = author;
+  commentItemDate.innerHTML = (commentDate.getMonth() + 1) + "/" 
+    + commentDate.getDate() + "/" 
+    + commentDate.getFullYear();
+  commentItemTime.innerHTML = commentDate.getHours() + ":" 
+    + commentDate.getMinutes() + ":"
+    + commentDate.getSeconds();
```
Comment (brucezheng) [inline portfolio/src/main/webapp/script.js]: When line wrapping, line break should come _after_ the operator.

https://google.github.io/styleguide/jsguide.html#formatting-where-to-break

**Review by brucezheng — APPROVED** [review]
Good job! Just have a few formatting issues to resolve.

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    HashMap<String, String> loginInfo = new HashMap<String, String>();
+    loginInfo.put("loginStatus", Boolean.toString(userService.isUserLoggedIn()));
+    loginInfo.put("loginURL", userService.createLoginURL("/index.html"));
+    loginInfo.put("logoutURL", userService.createLogoutURL("/index.html"));
+
+    response.setContentType("application/json;");
+    response.getWriter().println((new Gson()).toJson(loginInfo));
+  }
+
+}
```
Comment (ccyang314) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java]: nit: add a new line at the end of each new file.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -56,6 +57,44 @@ p {
   color: white;
 }
 
+#login {
+  background-color: #F4EBD9;
```
Comment (ccyang314) [inline portfolio/src/main/webapp/style.css]: nit: Looks like some color hex code are uppercase and some of them are lower cases. Could you make them consistent?

**Review by ccyang314 — CHANGES_REQUESTED** [review]

**Review by ccyang314 — APPROVED** [review]
LGTM

**Pull request merged**